cookbooks/bach_spark/recipes/default.rb

```diff
@@ -11,6 +12,35 @@
   mode 0755
 end
 
+template "#{spark_bin_dir}/conf/spark-defaults.conf" do
+  source "spark-defaults.conf.erb"
+  mode 0755
+end
+
+bash "create-hdfs-spark-history-dir" do
+  code <<-EOH
+  hdfs dfs -mkdir -p #{hdfs_url}/spark-history/
+  hdfs dfs -chmod -R 1777 #{hdfs_url}/spark-history/
+  hdfs dfs -chown -R hdfs:hdfs #{hdfs_url}/spark-history
+  EOH
+  user "hdfs"
+  not_if "sudo -u hdfs hadoop dfs -test -d #{hdfs_url}/spark-history"
+end
+
+bash "create-hdfs-spark-assembly-jar" do
+  code <<-EOH
+  hdfs dfs -mkdir -p #{hdfs_url}/apps/spark/#{spark_pkg_version}/
+  hdfs dfs -chown -R hdfs:hdfs #{hdfs_url}/apps/spark/#{spark_pkg_version}/
+  hdfs dfs -copyFromLocal  #{spark_bin_dir}/lib/spark-assembly-*.jar #{hdfs_url}/apps/spark/#{spark_pkg_version}/spark-assembly.jar
+  EOH
+  user "hdfs"
+  not_if "sudo -u hdfs hadoop dfs -test -d #{hdfs_url}/apps/spark/#{spark_pkg_version}/"
```

Comment:
Do you need to `sudo` as `hdfs` user if `bash` resource is running as `hdfs` user?